Added GSQL lexer

DanBarkus opened the pull request:
Added lexer for GSQL language and handling .gsql files

Anteru commented on pygments/lexers/gsql.py:
```diff
@@ -0,0 +1,78 @@
+            (r'.*\/\*\s*.*\s*\*\/', Comment.Multiline),
+        ],
+        'keywords': [
+            (r'(ACCUM|AND|ANY|API|AS|ASC|AVG|BAG|BATCH|BETWEEN|BOOL|BOTH|'
```
Please use the `words` function for long lists of keywords which can optimize the matching, see for instance: https://github.com/pygments/pygments/blob/master/pygments/lexers/python.py#L202

Anteru commented on pygments/lexers/gsql.py:
```diff
@@ -0,0 +1,78 @@
+
+__all__ = ["GSQLLexer"]
+
+class GSQLLexer(RegexLexer):
```
Please add a comment describing this lexer -- ideally with a link to the language definition. This comment should end with `.. versionadded:: 2.10`

zrougamed replied:
@DanBarkus thanks for the great work :smile: 
@Anteru I can't wait to see this PR Merged :smile:

GenericP3rson replied:
@DanBarkus Very cool! Can't wait to use it!
@Anteru Looking forward to the merge!

DanBarkus commented:
All requested changes have been made.

HerkTG commented:
@DanBarkus  thanks for completing this!!
@Anteru It would be great if you could review. We are looking to use this for an O'Reilly Book. Please review as soon as possible. Thanks!

Anteru commented on pygments/lexers/gsql.py:
```diff
@@ -0,0 +1,85 @@
+             'bitwiseoraccum', 'groupbyaccum', 'heapaccum', 'listaccum', 'MapAccum', 
+             'maxaccum', 'minaccum', 'oraccum', 'setaccum', 'sumaccum')), Name.Builtin),
+        ],
+        # 'name': [
+        #     (r'(\@\@w+)\b', Name),
+        # ],
```
Can you please remove this?

Anteru commented:
Merged, thanks! I'd appreciate if you could do a follow up commit to trim the example file a bit and reply to the question about `name` being commented out.

Anteru commented:
I also had to regenerate the golden test files, could you please check that `Name.Builtin` is acceptable for things like `FOR` which were `Keyword` previously?

Anteru commented:
@DanBarkus This is also failing the docs build as it's missing the top-level comment which includes the copyright annotation. I'm reverting the PR for now until this is addressed. Please reopen once this is resolved.

Anteru commented:
Reverted via 0a93d6c8ebe20d0e90883f210cd2db09f990ce0d and 066bd7e5f4bf864a3deac5994fa37f79801b2d23

DanBarkus commented:
I've added the missing copyright annotation and removed the unnecessary commented out section of code.
I reviewed uses of the words function in other modules (sql, python) and usage patterns line up with those languages.
Will submit new pull request with updated code.
